src/modules/poolModule.ts

```diff
@@ -983,4 +984,37 @@ export class PoolModule implements BaseModule {
       console.error(e);
     }
   }
+
+  public async fetchRoute(sourceToken: string, targetToken: string, amount: bigint) {
+    const extendedPools = await this._sdk.Pool.getAllPools();
```

Comment:
Allow use pools and tokens as input parameter to save request counts. If missing, go fetch.